XO.py

```diff
@@ -0,0 +1,157 @@
+from random import randint
+from collections import namedtuple
+
+# Size of board.
+N = 3
+
+
+class Model:
+    def __init__(self):
+        self.board = [[None for x in range(N)] for y in range(N)]
+
+    def get_board_status(self):
+        return self.board
+
+    def update_board(self, line, col, players, curr_player):
+        if self.board[line][col] is not None:
+            return False
+        else:
+            self.board[line][col] = players[curr_player].symbol
+            return True
+
+    def line_won(self, players, curr_player):
+        for line in self.board:
+            if all(x == players[curr_player].symbol for x in line):
+                return True
+        return False
+
+    def col_won(self, players, curr_player):
+        for col in range(N):
+            if all(line[col] == players[curr_player].symbol for line in self.board):
+                return True
+        return False
+
+    def diagonal_won(self, players, curr_player):
+        # Main diagonal.
+        if all(self.board[index][index] == players[curr_player].symbol for index in range(N)):
+            return True
+        # Opposite diagonal.
+        if all(self.board[index][N-1-index] == players[curr_player].symbol for index in range(N)):
+            return True
+        return False
+
+
+class View:

```

Comment:
Nice! But add a docstring for each class.

Reply:
Done